app/src/main/java/com/example/todo_list/EventsAdapter.kt

```diff
@@ -28,14 +27,13 @@ class EventsAdapter(val db : DataBase) : RecyclerView.Adapter<EventsAdapter.View
     override fun onBindViewHolder(holder: EventsAdapter.ViewHolder, position: Int) {
         val task = db.getTaskById(position)
```

Comment:
```suggestion
        val task = db.getTaskById(position + 1)
```
Всё как @Akrek и сказал. Просто _RecyclerView_ считает элементы от нуля, а _sqlite_ - от единицы.